CONTRIBUTING.md

```diff
@@ -1,7 +1,188 @@
+    ```
+
+4.  There was no specific issue being fixed here, but GitHub's squash-merge feature automatically appended `(#1177)`,
+    telling us which pull request created this commit. There's no description.
+    ```
+    Add "remember this option" checkbox to NVidia Optimus dialog (#1177)
+    ```
+
+5.  Here we refer to a previous commit.
+    ```
+    Addendum to a80f8d6: fix Windows build error
+    ```
+
+## Reviewing code
+
+Contributors should try to review other contributor's commits and provide
+feedback as much as possible.
+
+Please read our ["Code Review"] article for information on how to review code effectively.
+
+["Code Review"]: https://github.com/multitheftauto/mtasa-blue/wiki/Code-Review
+
+<!--
+
+TODO(qaisjp): the below content should be part of a code of conduct instead
+
+Ratings and comments are open for the public to review code and provide
+feedback. Please be mature and civilised when posting comments.
+
+Make sure you make appropriate use of the GitHub Reactions feature to
+rate commits or express agreement/disagreement to a comment. This avoids
+spammy comments such as "+1", "-1", "Nice one!", etc.
+
+Since you can only react to comments, not commits, feel free to create
+the initial "+1" comment in response to a commit. However, future
+similar reactions to a commit should be to the first response comment.
+
+-->
+
+## Gaining and losing merge rights
+
+Merge rights allow you to merge your own approved pull requests and 
+review other people's pull requests.
+
+We grant merge rights after you have proven yourself to be competent,
+which is generally after 3-5 pull requests. This is not fixed and depends
+on the extent of your contributions, community status and other factors.
+
+The subject matter of your pull requests do not matter — we are more interested in,
+once granted merge rights, whether you are capable of maintaining
+a high standard of code and remaining cohesive with other project collaborators.
+
+After gaining merge rights, if your contributions are of a consistently low standard,
+or you fail to stick to the rules, your permissions will be revoked.
+
+## Merging pull requests
+
+Before merging, enforced by GitHub's branch protection, pull requests **require**:
+- Linux and Windows status checks to pass
+- 1 pull request review
+
+If the pull request is large, try and only merge if there at least 2 pull request reviews.
+This isn't enforced via branch protection, but please try and stick to this convention
+(... unless nobody else is reviewing your PR).
+
+Branch protection is **not enforced** for repository administrators,
+and those people are therefore not required to send pull requests. Individual repository admins may,
+for the greater good, pledge to submit pull requests despite this lack of enforcement.
+
+For informational purposes, the current repository administrators are a _subset of_
+[those listed here](https://forum.mtasa.com/staff/)—it is not an authoritative list.
```

Comment:
I am summarising comments here from a private channel until the relevant parties send their PR review:
- `-1` one collaborator is not in favour of losing their own direct commit access
- `+1` one collaborator does not mind losing their own direct commit access, and appears to be in support <!-- pj -->
- `··` two inactive collaborators that would lose direct commit access, appear to abstain <!-- lopezloo, town -->
- `-1` one repo admin does not think existing permissions should change <!-- ccw -->
- `-1` one repo admin is not in favour of enforcing branch protection due to risk of "[not having] people reviewing pull requests on a timely basis" <!-- sbx -->
- `+2` two repo admins appear to be in support (myself included) <!-- qais, woovie -->
- `··` the remaining six repo admins, three of which are inactive, appear to abstain <!-- jusonex, botder, saml1er. inactive: caz, dan, ijsf -->

So at this time of this comment, it sums up to `+3 -3`. To get things moving, I've **drafted** the following suggestion in case branch protection is unwanted.

Please send improvements to this suggestion, in the comments, in the form of a GitHub _suggested change_.

```suggestion
Before merging, pull requests _should_ have:
- Linux and Windows status checks passing, and
- 1 pull request review, or 2 reviews for large pull requests

These rules are not enforced via branch protection but, for the greater good,
please try and stick to this convention (... unless nobody else is reviewing your PR).
```